catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/FeedbackApplication.java

```diff
@@ -0,0 +1,264 @@
+        this.emailDestination = emailDestination;
+    }
+
+    public void refresh(Map<String, Object> configuration) {
+        if (MapUtils.isEmpty(configuration)) {
+            return;
+        }
+
+        SmtpClient smtpClient = (SmtpClient)configuration.get("smtpClient");
+        if (smtpClient != null) {
+            this.smtpClient = smtpClient;
+        }
+
+        this.emailDestination = configurationApplication.getQueryFeedbackEmailDestination();
+        initTemplates(true);
+    }
+
+    private static FeedbackRequest parseFeedbackRequest(String json) {
+        FeedbackRequest feedbackRequest = new FeedbackRequest();
+
+        Map<String, Object> rootObject = MAPPER.parser()
+                .parseMap(json);
+
+        Map<String, Object> userObject = (Map<String, Object>) rootObject.get("user");
+        String name = (String) userObject.get("name");
+        String email = (String) userObject.get("email");
+
+        Map<String, Object> searchObject = (Map<String, Object>) rootObject.get("search");
+        String searchStr = (String) searchObject.get("cql");
+        Date searchInitiated = (Date) searchObject.get("initiated");
+        Collection<String> searchResults = (Collection<String>) searchObject.get("results");
+        ValueList searchStatus = (ValueList) searchObject.get("status");
+
+        Map<String, Object> workspaceObject = (Map<String, Object>) rootObject.get("workspace");
+        String workspaceId = (String) workspaceObject.get("id");
+        String workspaceName = (String) workspaceObject.get("name");
+
+        String comments = (String) rootObject.get("comments");
+
+        feedbackRequest.setUsername(name);
+        feedbackRequest.setEmail(email);
+        feedbackRequest.setQuery(searchStr);
+        feedbackRequest.setQueryInitiated(searchInitiated.toString());
+        feedbackRequest.setQueryResults(searchResults.stream()
+                .map(Object::toString)
+                .collect(Collectors.joining(", ")));
+        feedbackRequest.setQueryStatus(searchStatus.stream()
+                .map(Object::toString)
+                .collect(Collectors.joining(", ")));
+        feedbackRequest.setWorkspaceId(workspaceId);
+        feedbackRequest.setWorkspaceName(workspaceName);
+        feedbackRequest.setComments(comments);
+
+        return feedbackRequest;
+    }
+
+    private String getEmailSubject(FeedbackRequest feedback) {
+        String subject = null;
+
+        initTemplates(false);
+
+        if (subjectTemplate != null) {
+            Map<String, String> templateValueMap = getTemplateValueMap(feedback);
+            try {
+                subject = subjectTemplate.apply(templateValueMap);
+            } catch (IOException e) {
+                LOGGER.error("Unable to apply values to handle bars email subject template", e);
+            }
+        }
+
+        return subject;
+    }
+
+    private String getEmailBody(FeedbackRequest feedback) {
```

Comment:
This method is nearly identical to getEmailSubject. Can you refactor them to remove the common code?